### PR TITLE
Fix lib install with git url

### LINKED ISSUE #1120: lib install --git-url with a local path does not work as expected
### Current behavior

```
$ arduino-cli lib install --git-url /path/to/a/local/MyLibrary
--git-url and --zip-path flags allow installing untrusted files, use it at your own risk.
Enumerating objects: 305, done.
Counting objects: 100% (305/305), done.
Compressing objects: 100% (159/159), done.
Total 305 (delta 132), reused 280 (delta 126)
Installed Library from Git URL
```

It looks like a local path is accepted as argument for `--git-url`. However the library is installed incorrectly, because it looks like it is copied in the root of the libraries directory instead of under its own directory:

```
ls ~/Documents/Arduino/libraries
README.adoc			docs			examples		keywords.txt		library.properties	src
```

### Expected behavior

Either:

```
ls ~/Documents/Arduino/libraries
MyLibrary
```

or:

```
Error: invalid URL supplied.
```

Note that it works correctly if I supply a proper git URL (`https://github.com/foo/bar.git`).

### Environment

- CLI version (output of `arduino-cli version`): `arduino-cli alpha Version: 0.14.0 Commit: a86b21d99e2af9e0857da0ce4ab80baf1d3afb55 Date:`
- OS and platform: `macOS`

### Additional context

I was looking for a way to install a library located in my filesystem. A `--path` argument could be handy in addition to `--git-url` and `--zip-file`. Incidentally, my local library is a clone of a git repo so I tried to use `--git-url` so I discovered this bug.

### PR DESCRIPTION
**Pull request by silvanocerza**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**

Fixes a bug with an existing command.

- **What is the current behavior?**

Calling `arduino-cli lib install --git-url .` is executed successfully but the library is installed in the wrong folder, to be precise it's installed directly inside the `<sketch_book_dir>/libraries` folder, so the whole content of the library ends there.

* **What is the new behavior?**

Calling `arduino-cli lib install --git-url .` now correctly deduces the library name from the given path and installs it in the `<sketch_book_dir>/libraries/<lib_name>` folder.

- **Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**

Nope.

* **Other information**:

Fixes #1120

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

File: arduino/libraries/librariesmanager/install.go [hunk 1]
```diff
@@ -111,21 +112,42 @@ func (lm *LibrariesManager) InstallZipLib(ctx context.Context, archivePath strin
...
 	if err != nil {
 		return err
 	}
 	return nil
 }
+
+func parseGitURL(gitURL string) (string, error) {
+	var res string
+	if strings.HasPrefix(gitURL, "git") || strings.HasPrefix(gitURL, "ssh") {
+		// We can't parse these as URLs
```
Comment (per1234) [inline arduino/libraries/librariesmanager/install.go:141]: Is this for the `git://github.com/foo/bar` and `ssh://git@github.com/foo/bar.git` style of URLs? `url.Parse()` seem to handle them fine in my quick experiments.

Comment (silvanocerza) [reply arduino/libraries/librariesmanager/install.go:141]: Nope, this is about URLs like `git@github.com:arduino/arduino-cli.git`. I could probably change this to `strings.HasPrefix(gitURL, "git@")`. :thinking: 

I'll give it a try.

File: test/test_lib.py [hunk 2]
```diff
@@ -383,3 +387,77 @@ def test_lib_list_with_updatable_flag(run_command):
...
+    lib_install_dir = Path(data_dir, "libraries", "WiFi101")
+    # Verifies library is not installed
+    assert not lib_install_dir.exists()
+
+    # Clone repository locally
+    git_url = "https://github.com/arduino-libraries/WiFi101.git"
+    repo_dir = Path(data_dir, "WiFi101")
+    assert Repo.clone_from(git_url, repo_dir)
+
+    assert run_command(f"lib install --git-url {repo_dir.as_uri()}", custom_working_dir=repo_dir, custom_env=env)
```
Comment (per1234) [inline test/test_lib.py:441]: Is there a reason for the `custom_working_dir=repo_dir`? It seems like this is only necessary in the `test_install_with_git_url_from_current_directory` test.

**Review by per1234 — CHANGES_REQUESTED** [review]
It would be nice to have a test for a relative path to cover all the use cases.

File: arduino/libraries/librariesmanager/install.go (hunk 1 above)
Comment (per1234) [reply arduino/libraries/librariesmanager/install.go:141]: Thanks. I think that's a nice improvement.

File: test/test_lib.py (hunk 2 above)
Comment (silvanocerza) [reply test/test_lib.py:441]: Yep, copy and pasting strikes again. :grin:

**Review by ubidefeo — APPROVED** [review]
tested and confirmed working

**Review by per1234 — APPROVED** [review]

**Pull request merged**